src/provider/provider_os_memory.c

```diff
@@ -1251,14 +1247,14 @@ static umf_result_t os_open_ipc_handle(void *provider, void *providerIpcData,
     umf_result_t ret = UMF_RESULT_SUCCESS;
     int fd;
 
-    if (os_provider->shm_name[0]) {
-        fd = utils_shm_open(os_provider->shm_name);
+    if (os_ipc_data->shm_name[0]) {
```

Comment:
Is the memory zeroed on the producer side? This check make sense only if the `shm_name` string is zeroed on the producer side.

Reply:
yes, see L#245

Reply:
Another question, I see that you returned size optimization when shm_name is not used. In that case, we cannot check `os_ipc_data->shm_name[0]` because it is out of bound of the `os_ipc_data` structure.

Reply:
But the in case shm is not used we still can check the first byte - if it is set to 0 then we don't read the rest

Reply:
There should be added `+1`:
```c
        // os_ipc_data_t->shm_name will NOT be used
        *size = sizeof(os_ipc_data_t) - NAME_MAX + 1;
```
so that the first byte is accessible.
And later in the `os_get_ipc_handle()`:
```c
    if (os_provider->shm_name[0]) {
        strncpy(os_ipc_data->shm_name, os_provider->shm_name, NAME_MAX - 1);
        os_ipc_data->shm_name[NAME_MAX - 1] = '\0';
    } else {
        os_ipc_data->fd = os_provider->fd;
        os_ipc_data->shm_name[0] = '\0'; // <<<<<<<<<<<<<<<< THIS LINE SHOULD BE ADDED
    }

```

Reply:
I implemented this as Łukasz D suggested - this is potentially less error-prone in the future